aten/src/ATen/xpu/profiler/XPUActivityProfiler.cpp

```diff
@@ -0,0 +1,178 @@
+}
+
+// =========== Session Public Methods ============= //
+void XPUActivityProfilerSession::start() {
+  profilerStartTs_ =
+      libkineto::timeSinceEpoch(std::chrono::high_resolution_clock::now());
+}
+
+void XPUActivityProfilerSession::stop() {
+  xpti_.disablePtiActivities(activity_types_);
+  profilerEndTs_ =
+      libkineto::timeSinceEpoch(std::chrono::high_resolution_clock::now());
+}
+
+void XPUActivityProfilerSession::processTrace(logger_t& logger) {
+  traceBuffer_.span = libkineto::TraceSpan(
+      profilerStartTs_, profilerEndTs_, "__xpu_profiler__");
+  traceBuffer_.span.iteration = iterationCount_++;
+  auto gpuBuffer = xpti_.activityBuffers();
+  if (gpuBuffer) {
+    xpti_.processActivities(
+        *gpuBuffer,
+        std::bind(
+            &XPUActivityProfilerSession::handlePtiActivity,
+            this,
+            std::placeholders::_1,
+            &logger));
+  }
+}
+
+void XPUActivityProfilerSession::processTrace(
+    logger_t& logger,
+    libkineto::getLinkedActivityCallback get_linked_activity,
+    int64_t captureWindowStartTime,
+    int64_t captureWindowEndTime) {
+  captureWindowStartTime_ = captureWindowStartTime;
+  captureWindowEndTime_ = captureWindowEndTime;
+  cpuActivity_ = get_linked_activity;
+  processTrace(logger);
+}
+
+std::unique_ptr<libkineto::DeviceInfo> XPUActivityProfilerSession::
+    getDeviceInfo() {
+  return {};
+}
+
+std::vector<libkineto::ResourceInfo> XPUActivityProfilerSession::
+    getResourceInfos() {
+  return {};
+}
+
+std::unique_ptr<libkineto::CpuTraceBuffer> XPUActivityProfilerSession::
+    getTraceBuffer() {
+  return std::make_unique<libkineto::CpuTraceBuffer>(std::move(traceBuffer_));
+}
+
+void XPUActivityProfilerSession::pushCorrelationId(uint64_t id) {
+  xpti_.pushCorrelationID(id, XPUActivityApi::CorrelationFlowType::Default);
+}
+
+void XPUActivityProfilerSession::popCorrelationId() {
+  xpti_.popCorrelationID(XPUActivityApi::CorrelationFlowType::Default);
+}
+
+void XPUActivityProfilerSession::pushUserCorrelationId(uint64_t id) {
+  xpti_.pushCorrelationID(id, XPUActivityApi::CorrelationFlowType::User);
+}
+
+void XPUActivityProfilerSession::popUserCorrelationId() {
+  xpti_.popCorrelationID(XPUActivityApi::CorrelationFlowType::User);
+}
+
+void XPUActivityProfilerSession::enumDeviceUUIDs() {
+  auto platform_list = sycl::platform::get_platforms();
+  // Enumerated GPU devices from the specific platform.
+  for (const auto& platform : platform_list) {
+    if (platform.get_backend() != sycl::backend::ext_oneapi_level_zero) {
+      continue;
+    }
+    auto device_list = platform.get_devices();
+    for (const auto& device : device_list) {
+      if (device.is_gpu()) {
+        if (device.has(sycl::aspect::ext_intel_device_info_uuid)) {
+          deviceUUIDs_.push_back(
+              device.get_info<sycl::ext::intel::info::device::uuid>());
+        } else {
+          std::cerr << "UUID is not supported for this XPU device" << std::endl;
```

Comment:
not sure if std::cerr is a good solution.

Reply:
because how std throws warning, i only find std::runtime_error

Reply:
Let's warn user, instead of giving an error.

Reply:
We can move similar logic of TORCH_CHECK here.

Reply:
We mistaken here. We need to reference the `TORCH_WARN` to give the warnings. The `TORCH_CHECK` will throw the error.

Reply:
`TORCH_WARN` is a little bit complicated. For now i use std::cerr.
We can have a better solution here for cpp easy warning infra.